### PR TITLE
Add license and changelog files to the UPM packages

### PR DESCRIPTION
**Pull request by davidkline-ms**
Per #8734, our packages were missing license and changelog files.

This change adds these files and updates the packaging process to ensure they are included.

This change also adds the Documentation~ folder to each package containing a readme with links to the docs.microsoft.com MRTK getting started as well as the MRTK docs on GitHub.

Fixes: #8734

### CODE REVIEW

File: Assets/MRTK/packagetemplate.json [hunk 1]
```diff
@@ -32,6 +32,9 @@
         "SDK*",
         "Services*",
         "package.json.meta",
+        "LICENSE*",
```
Comment (keveleigh) [inline Assets/MRTK/packagetemplate.json:35]: Do we need these in all the other packages too? I think I just see Foundation and Examples

File: NOTICE.md.meta [hunk 2]
```diff
@@ -0,0 +1,7 @@
+fileFormatVersion: 2
+guid: 7eac9b4c9a2ab1f40aab89ba9a3b66a6
```
Comment (keveleigh) [inline NOTICE.md.meta:2]: We'll probably need unique metas for each package this gets copied to. Otherwise there will be GUID collisions when multiple packages are installed.

File: Assets/MRTK/packagetemplate.json (hunk 1 above)
Comment (keveleigh) [reply Assets/MRTK/packagetemplate.json:35]: At least LICENSE and maybe CHANGELOG?

Comment (davidkline-ms) [reply Assets/MRTK/packagetemplate.json:35]: The other packages are governed by .npmignore files. They will get picked up automatically

File: NOTICE.md.meta (hunk 2 above)
Comment (davidkline-ms) [reply NOTICE.md.meta:2]: good thought.... i suppose we should just check the files in to each package root.

File: scripts/packaging/createupmpackages.ps1 [hunk 3]
```diff
@@ -83,6 +83,7 @@ $packages = [ordered]@{
 # Beginning of the upm packaging script main section
 # The overall structure of this script is:
 #
+# 0) Ensure necessary documentation files (license, changelog, etc.) are copied to the appropriate location
```
Comment (wiwei) [inline scripts/packaging/createupmpackages.ps1:86]: Nit: If we want to avoid numbering things, it would probably be easier to replace the numbers with bullet points so that we don't have to do highway exit ordering in the future

**Review by wiwei — COMMENTED** [review]
Why are there .meta files for the markdown things (i.e. CHANGELOG/LICENSE/NOTICE) in the root directory (i.e. usually there aren't meta files associated with that).

**Comment by davidkline-ms** [comment]
> Why are there .meta files for the markdown things (i.e. CHANGELOG/LICENSE/NOTICE) in the root directory

They were needed to satisfy Unity's requirement of having meta files in the packages. As @keveleigh pointed out, we should have unique metas for each copy. These files are now maintained in the root and copied to the package folders as part of the upm script.

File: Assets/MRTK/Examples/Experimental/ExamplesHub/Profiles/MRTKExamplesHubSceneSystemProfile.asset [hunk 4]
```diff
@@ -16,7 +16,7 @@ MonoBehaviour:
   useManagerScene: 0
   managerScene:
     Name: DefaultManagerScene
-    Path: Assets/MRTK/SceneSystemResources/DefaultManagerScene.unity
+    Path: Assets/MRTK/Services/SceneSystem/SceneSystemResources/DefaultManagerScene.unity
```
Comment (keveleigh) [inline Assets/MRTK/Examples/Experimental/ExamplesHub/Profiles/MRTKExamplesHubSceneSystemProfile.asset:19]: Is this change intended for this PR?

File: Assets/MRTK/NOTICE.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
+NOTICES AND INFORMATION
```
Comment (keveleigh) [inline Assets/MRTK/NOTICE.md:1]: Instead of having multiple copies of each of the .md files (which could get unwieldy when we need to update the contents in all places), can we just have multiple copies of the .meta files that get copied in during packaging? And copy the existing .md files from the root for all packages

Comment (keveleigh) [reply Assets/MRTK/NOTICE.md:1]: In a different project, I had a UnityMetas folder in the CI/packaging scripting folder and copied then in when relevant. We could have something like NOTICE.md.meta.examples and `Copy-Item -Path "$projectRoot/packaging/UnityMetas/LICENSE.md.meta.$entry" -Destination "$packagePath/LICENSE.md.meta"` or wherever

File: Assets/MRTK/Examples/Experimental/ExamplesHub/Profiles/MRTKExamplesHubSceneSystemProfile.asset (hunk 4 above)
Comment (davidkline-ms) [reply Assets/MRTK/Examples/Experimental/ExamplesHub/Profiles/MRTKExamplesHubSceneSystemProfile.asset:19]: this looks like it got missed from a different PR? this is the current location for the files.

Comment (davidkline-ms) [reply Assets/MRTK/Examples/Experimental/ExamplesHub/Profiles/MRTKExamplesHubSceneSystemProfile.asset:19]: maybe it's GH being confused?

File: Assets/MRTK/NOTICE.md (hunk 5 above)
Comment (davidkline-ms) [reply Assets/MRTK/NOTICE.md:1]: i'll do something like that

File: Assets/MRTK/Examples/Experimental/ExamplesHub/Profiles/MRTKExamplesHubSceneSystemProfile.asset (hunk 4 above)
Comment (keveleigh) [reply Assets/MRTK/Examples/Experimental/ExamplesHub/Profiles/MRTKExamplesHubSceneSystemProfile.asset:19]: Hrm yeah, strange. It's in this commit for some reason https://github.com/microsoft/MixedRealityToolkit-Unity/pull/8786/commits/8cc16e7e3edb13f1742e141e76779ba65142ee1c

Comment (davidkline-ms) [reply Assets/MRTK/Examples/Experimental/ExamplesHub/Profiles/MRTKExamplesHubSceneSystemProfile.asset:19]: That is odd... It _is_ the correct path, however.

Comment (davidkline-ms) [reply Assets/MRTK/Examples/Experimental/ExamplesHub/Profiles/MRTKExamplesHubSceneSystemProfile.asset:19]: provided that nothing else changed in the profile (index, order, etc). this should be what we want.

File: Assets/MRTK/NOTICE.md (hunk 5 above)
Comment (davidkline-ms) [reply Assets/MRTK/NOTICE.md]: done

File: UPM/UnityMetaFiles/NOTICE.md.md.meta.tools [hunk 6]
```diff
@@ -0,0 +1,7 @@
+fileFormatVersion: 2
+guid: cdb0f77fe63ed9743ad6ce60ca803c3d
```
Comment (keveleigh) [inline UPM/UnityMetaFiles/NOTICE.md.md.meta.tools:2]: The name on this file has two `.md`s

File: scripts/packaging/createupmpackages.ps1 [hunk 7]
```diff
@@ -110,6 +111,12 @@ foreach ($entry in $packages.GetEnumerator()) {
     # Create and publish the package
     $packageName = $entry.Name
 
+    # Copy files used by UPM to display license, change log, etc.
+    Copy-Item -Path "$projectRoot/LICENSE.md" "$packagePath"
+    Copy-Item -Path "$projectRoot/NOTICE.md" "$packagePath"
+    Copy-Item -Path "$projectRoot/CHANGELOG.md" "$packagePath"
+    Copy-Item -Path "$projectRoot/packaging/UnityMetaFiles/$packageName/*.meta" "$packagePath"
```
[the remark this answers is not in the document]
Comment (keveleigh) [reply scripts/packaging/createupmpackages.ps1:118]: (I think I like generally the path represented here though, over a new root-level UPM folder just for metas)

[the remark this answers is not in the document]
Comment (keveleigh) [reply scripts/packaging/createupmpackages.ps1:118]: (ah, though packaging would be a new root-level folder too. nevermind!)

File: UPM/UnityMetaFiles/NOTICE.md.md.meta.tools (hunk 6 above)
Comment (davidkline-ms) [reply UPM/UnityMetaFiles/NOTICE.md.md.meta.tools:2]: caught it

File: scripts/packaging/createupmpackages.ps1 (hunk 7 above)
[the remark this answers is not in the document]
Comment (davidkline-ms) [reply scripts/packaging/createupmpackages.ps1:118]: fixing

**Review by keveleigh — APPROVED** [review]
